,dep,version,filepath,bloated,cve,affected versions,advisory
0,pytest,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/requirements.txt,False,no one,no one,no one
1,spacy,">=2.1.0,<3.0.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/requirements.txt,False,no one,no one,no one
2,cython,>=0.25,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/requirements.txt,False,no one,no one,no one
3,numpy,>=1.15.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2019-6446,<1.16.3,"Numpy 1.16.3 includes a fix for CVE-2019-6446: It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call.
https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
4,numpy,>=1.15.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2021-33430,<1.21.0rc1,"Numpy 1.21.0rc1 includes a fix for CVE-2021-33430: A Buffer Overflow vulnerability in the PyArray_NewFromDescr_int function of ctors.c when specifying arrays of large dimensions (over 32) from Python code, which could let a malicious user cause a Denial of Service. 
NOTE: The vendor does not agree this is a vulnerability. In (very limited) circumstances a user may be able provoke the buffer overflow, the user is most likely already privileged to at least provoke denial of service by exhausting memory. Triggering this further requires the use of uncommon API (complicated structured dtypes), which is very unlikely to be available to an unprivileged user.
https://github.com/numpy/numpy/issues/18939"
5,numpy,>=1.15.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2021-34141,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-34141: An incomplete string comparison in the numpy.core component in NumPy before 1.22.0 allows attackers to trigger slightly incorrect copying by constructing specific string objects. 
NOTE: the vendor states that this reported code behavior is ""completely harmless.""
https://github.com/numpy/numpy/issues/18993"
6,numpy,>=1.15.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2021-41496,<1.22.0,"Numpy 1.22.0 includes a fix for CVE-2021-41496: Buffer overflow in the array_from_pyobj function of fortranobject.c, which allows attackers to conduct a Denial of Service attacks by carefully constructing an array with negative values. 
NOTE: The vendor does not agree this is a vulnerability; the negative dimensions can only be created by an already privileged user (or internally).
https://github.com/numpy/numpy/issues/19000"
7,numpy,>=1.15.0,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2021-41495,<1.22.2,"Numpy 1.22.2  includes a fix for CVE-2021-41495: Null Pointer Dereference vulnerability exists in numpy.sort in NumPy in the PyArray_DescrNew function due to missing return-value validation, which allows attackers to conduct DoS attacks by repetitively creating sort arrays. 
NOTE: While correct that validation is missing, an error can only occur due to an exhaustion of memory. If the user can exhaust memory, they are already privileged. Further, it should be practically impossible to construct an attack which can target the memory exhaustion to occur at exactly this place.
https://github.com/numpy/numpy/issues/19038"
8,boto3,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,PVE-2021-41708,<1.4.5,"Boto3 version 1.4.5 fixes an information exposure vulnerability: The boto logger boto3.resources.action, which propagates to root logger, logs the entire uploaded bytes at INFO level.
https://github.com/boto/boto3/issues/1017"
9,requests,">=2.13.0,<3.0.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2018-18074,<=2.19.1,"Requests before 2.20.0 sends an HTTP Authorization header to an http URI upon receiving a same-hostname https-to-http redirect, which makes it easier for remote attackers to discover credentials by sniffing the network."
10,requests,">=2.13.0,<3.0.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2015-2296,">=2.1,<=2.5.3",The resolve_redirects function in sessions.py in requests 2.1.0 through 2.5.3 allows remote attackers to conduct session fixation attacks via a cookie without a host value in a redirect.
11,requests,">=2.13.0,<3.0.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2023-32681,">=2.3.0,<2.31.0","Requests is a HTTP library. Since Requests 2.3.0, Requests has been leaking Proxy-Authorization headers to destination servers when redirected to an HTTPS endpoint. This is a product of how we use 'rebuild_proxies' to reattach the 'Proxy-Authorization' header to requests. For HTTP connections sent through the tunnel, the proxy will identify the header in the request itself and remove it prior to forwarding to the destination server. However when sent over HTTPS, the 'Proxy-Authorization' header must be sent in the CONNECT request as the proxy has no visibility into the tunneled request. This results in Requests forwarding proxy credentials to the destination server unintentionally, allowing a malicious actor to potentially exfiltrate sensitive information. This issue has been patched in version 2.31.0.

"
12,spacy,">=2.1.0,<3.0.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,no one,no one,no one
13,wheel,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/setup.py,False,CVE-2022-40898,<0.38.1,"Wheel 0.38.1 includes a fix for CVE-2022-40898: An issue discovered in Python Packaging Authority (PyPA) Wheel 0.37.1 and earlier allows remote attackers to cause a denial of service via attacker controlled input to wheel cli.
https://pyup.io/posts/pyup-discovers-redos-vulnerabilities-in-top-python-packages"
14,torch,">=1.3.0,<1.4.0",/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/neuralcoref/train/training_requirements.txt,False,CVE-2022-45907,">=0,<1.13.1","Torch 1.13.1 includes a fix for CVE-2022-45907: In PyTorch before trunk/89695, torch.jit.annotations.parse_type_line can cause arbitrary code execution because eval is used unsafely.
https://github.com/pytorch/pytorch/issues/88868"
15,tensorboardX,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/neuralcoref/train/training_requirements.txt,False,no one,no one,no one
16,spacy,==*,/home/daniele/git/NICHE_projects/Repo/Cloneables/neuralcoref/neuralcoref/train/training_requirements.txt,False,no one,no one,no one
